### PR TITLE
Fix #777 deprecate sb

### LINKED ISSUE #777: Deprecate SB elements relative to msg module adaption
**Is your feature request related to a problem? Please describe.**
Many APIs will be updated for consistency relative to the MSG module, also element scoping improvements (SB doesn't actually care about the header, it just needs to route).

**Describe the solution you'd like**
Per 2020-07-28 discussion SB for once #726 is in.

Deprecating:
- [x] CFE_SB_PKTTYPE_* -> CFE_MSG_Type_t
- [x] CFE_SB_MsgPtr_t -> CFE_MSG_Message_t *
- [x] CFE_SB_Msg_t -> CFE_MSG_Message_t
- [x] CFE_SB_MsgPayloadPtr_t (use pointer to payload in structure)
- [x] CFE_SB_InitMsg -> CFE_MSG_Init
- [x] CFE_SB_GetTotalMsgLength -> CFE_MSG_GetSize
- [x] CFE_SB_SetTotalMsgLength -> CFE_MSG_SetSize
- [x] CFE_SB_GetMsgTime -> CFE_MSG_GetMsgTime (this gets rid of structure return, similar to #45 issue)
- [x] CFE_SB_SetMsgTime -> CFE_MSG_SetMsgTime
- [x] CFE_SB_GetCmdCode -> CFE_MSG_GetFcnCode
- [x] CFE_SB_SetCmdCode -> CFE_MSG_SetFcnCode
- [x] CFE_SB_GetChecksum (no use case defined, what do you need it for?)
- [x] CFE_SB_GenerateChecksum -> CFE_MSG_GenerateChecksum
- [x] CFE_SB_ValidateChecksum -> CFE_MSG_ValidateChecksum
- [x] CFE_SB_GetMsgId -> CFE_MSG_GetMsgId
- [x] CFE_SB_SetMsgId -> CFE_MSG_SetMsgId
- [x] CFE_SB_GetPktType -> CFE_MSG_GetTypeFromMsgId
- [x] CFE_SB_SetMsgSeqCnt -> CFE_MSG_SetSequenceCount

NOT deprecating, but will note in API that these are fragile (guesses based on assumptions).  Future implementation could be replaced via data dictionary sort of access or standards based header definitions (including secondary header, and flags or extra internal data to mange the real sizes):
- CFE_SB_MsgHdrSize: use actual message structure where possible
- CFE_SB_GetUserData: use actual message structure where possible
- CFE_SB_GetUserDataLength: use actual message structure where possible
- CFE_SB_SetUserDataLength - use CFE_MSG_SetSize with full message structure where possible
- CFE_SB_CMD_HDR_SIZE -> sizeof CFE_MSG_CommandHeader_t) preferrred
- CFE_SB_TLM_HDR_SIZE -> sizeof (CFE_MSG_TelemetryHeader_t) preferred

Note CFE_SB_Qos_t and CFE_SB_Default_Qos will likely be used for #926 (critical subscription)

NOT deprecating, MSG types are unaligned, SB types are now aligned:
- CFE_SB_Msg_t, CmdHdr_t, TlmHdr_t -> CFE_MSG_*

NOT deprecating CFE_SB_TimeStampMsg.  See #26 that requests making this part of send.

**Describe alternatives you've considered**
Just need to manage these, unique deprecation flag... not all actually need to go, but reduced/simplifies unit testing

**Additional context**
#711, #726

**Requester Info**
Jacob Hageman - NASA/GSFC

### PR DESCRIPTION
**Pull request by skliper**
**Describe the contribution**
Fix #777 - see issue for deprecated elements

**Testing performed**
Built and ran, enabled telemetry, checked noops.  Unit tests built and passed.

**Expected behavior changes**
None

**System(s) tested on**
 - Hardware: cFS Dev Server
 - OS: Ubuntu 18.04
 - Versions: Bundle main + this commit

**Additional context**
nasa/cFE#777

**Third party code**
None

**Contributor Info - All information REQUIRED for consideration of pull request**
Jacob Hageman - NASA/GSFC

### CODE REVIEW

**Comment by astrogeco** [comment]
[CCB 2020-11-04](https://github.com/nasa/cFS/wiki/Community-CCB-Agenda:-November-04-2020) 
- Replaces old refrences with new ones. Also updated the App dev guide language. 
- Tables also use the new APIs
- Hope to review asynchronously or have a splinter

**Comment by skliper** [comment]
Ready for review - 8514efa unit test updates, 1da6b3c doc updates, 34e5510 core updates.

File: fsw/cfe-core/src/evs/cfe_evs_task.c [hunk 1]
```diff
@@ -212,8 +212,8 @@ int32 CFE_EVS_CleanUpApp(CFE_ES_ResourceID_t AppID)
 */
 void CFE_EVS_TaskMain(void)
 {
-    int32 Status;    
-    CFE_SB_MsgPtr_t    EVS_MsgPtr; /* Pointer to SB message */
+    int32              Status;
+    CFE_MSG_Message_t *MsgPtr; /* Pointer to SB message */
```
Comment (zanzaben) [inline fsw/cfe-core/src/evs/cfe_evs_task.c:216]: Should this comment should be change to not mention SB?
```suggestion
    CFE_MSG_Message_t *MsgPtr; /* Pointer to message */
```

**Comment by skliper** [comment]
@acudmore @jphickey @ejtimmon Please review so I can address #1009 and move forward with the standard pattern implementation.

File: fsw/cfe-core/src/evs/cfe_evs_task.c (hunk 1 above)
Comment (skliper) [reply fsw/cfe-core/src/evs/cfe_evs_task.c:216]: Planning to address as part of #1009 cleanup.

**Review by jphickey — COMMENTED** [review]
I've reviewed, and I'm honestly a little reluctant to go forth with this - we definitely have more to discuss here!  I'm really not convinced we should be removing the `CFE_SB_Msg_t` type.  There is an important distinction between a message in its raw form and a message as its being transported via SB - like the analogy of a letter vs. an envelope going through the post office to be delivered to its destination.  MSG lib should be dealing with letters while SB deals with envelopes.  They are related, but different.

My concern is that this change set is _undoing_ a lot of that distinction that was already there.  I suppose we can put it back in #1009, or maybe we should hold off on at least the final commit in this change set.

FWIW - I am in agreement with the removal of the "clear" flag on init (first commit) and the removal of most "manipulator" routines - again with the analogy of an envelope, once it is "sealed" you can look at its basic stuff such as the address/delivery info (msgid, size) but not the stuff inside (timestamps, cmd codes, payload).  So it makes sense that the SB functions that get/set internal message fields should be deprecated in favor of the MSG versions.  But when dealing with messages in their delivery form we should still have an SB level API for that that works with SB defined types.

**Comment by jphickey** [comment]
> @acudmore @jphickey @ejtimmon Please review so I can address #1009 and move forward with the standard pattern implementation.

Sorry for the delay I was under the impression we might have an actual splinter meeting about this, but I'm fine to do it all via github as well.

File: fsw/cfe-core/src/evs/cfe_evs_utils.c [hunk 2]
```diff
@@ -428,7 +428,7 @@ void EVS_GenerateEventTelemetry(EVS_AppData_t *AppDataPtr, uint16 EventID, uint1
     LongEventTlm.Payload.PacketID.ProcessorID  = CFE_PSP_GetProcessorId();
 
     /* Set the packet timestamp */
-    CFE_SB_SetMsgTime((CFE_SB_Msg_t *) &LongEventTlm, *TimeStamp);
+    CFE_MSG_SetMsgTime((CFE_MSG_Message_t *) &LongEventTlm, *TimeStamp);
```
[the remark this answers is not in the document]
Comment (skliper) [reply fsw/cfe-core/src/evs/cfe_evs_utils.c:431]: Will address in #1009.

File: docs/cFE Application Developers Guide.md [hunk 3]
```diff
@@ -1945,10 +1950,9 @@ SAMPLE_AppData_t  SAMPLE_AppData;  /* Instantiate Task Data */
    ** Get a SB Message block of memory and initialize it
    */
    SAMPLE_AppData.BigPktPtr = (SAMPLE_BigPkt_t *)CFE_SB_ZeroCopyGetPtr(SAMPLE_BIGPKT_MSGLEN);
-   CFE_SB_InitMsg((CFE_SB_Msg_t *) SAMPLE_AppData.BigPktPtr,
-                                   SAMPLE_BIG_TLM_MID,
-                                   SAMPLE_BIGPKT_MSGLEN,
-                                   CFE_SB_CLEAR_DATA);
+   CFE_MSG_Init((CFE_MSG_Message_t *) SAMPLE_AppData.BigPktPtr,
+                                      SAMPLE_BIG_TLM_MID,
+                                      SAMPLE_BIGPKT_MSGLEN);
```
[the remark this answers is not in the document]
Comment (skliper) [reply docs/cFE Application Developers Guide.md:1955]: Will resolve in #1009

**Comment by skliper** [comment]
@astrogeco can we move forward with this one?  Maybe create a new issue for the documentation and fix it later.

**Pull request merged**